runner/src/Quickstrom/Run.hs

```diff
@@ -331,25 +341,39 @@ generateValidActions possibleActions = loop
             & (>>= Pipes.yield)
           loop
 
-selectValidAction :: (MonadIO m, WebDriver m) => Action Selector -> Runner m (Maybe (Action Selected))
-selectValidAction possibleAction =
+selectValidAction :: (MonadIO m, WebDriver m) => PotentialAction -> Runner m (Maybe Action)
+selectValidAction pa = traverseRest (zip ((Data.List.replicate 1 False) ++ (cycle [True])) pa) []
```

Comment:
This code will always run queries (e.g. `isActiveInput`) on actions, even if that result isn't used due to the action being in a sequence.

I think we can simplify this a lot by removing the zipping with bools in `traverseRest`, and just checking that the first action in the sequence is valid (using `selectValidBaseAction`), and selecting elements for the rest of the sequence. Something like:

```haskell
selectValidBaseAction (Vector.head pa) >>= \case
  Just firstAction -> do
    restActions <- traverse selectUnconditionally (Vector.tail pa)
    pure (Just (Vector.cons firstAction restActions))
  Nothing -> pure Nothing
```

Reply:
Sure, but `selectUnconditionally` doesn't exist. Are you suggesting to pull it out of `selectValidBaseAction` and call it directly on the tail elements? Yes, I think that's a bit cleaner but will require duplicating the pattern matching on the action types.

Reply:
Ah, right. Maybe you can tweak `selectValidBaseAction` to take a single bool that overrides the validation, sort of what you did with the tuple.

Ideally, we'd separate the "select" and "validate" phases. First we'd go `BaseAction Selector -> BaseAction [(Element, Selected)]`, and then `BaseAction [(Element, Selected)] -> Maybe  (BaseAction Selected)` after validation. But maybe let's keep that one for a separate PR.

Reply:
Not sure I follow. Isn't `(a,b) -> c` effectively the same as `a -> b -> c` ? If I change it to the latter, it complicates the zip logic.

If you accept the above reasoning, this PR is now ready for merge. If not, let me know and it will change from `(a,b) -> c` to `a -> b -> c`.

Reply:
But yes I agee with separating "select" and "validate" phases in principle & that it should be a followup PR.